src/libsyntax/parse/token.rs

```diff
@@ -122,7 +122,8 @@ pub(crate) fn ident_can_begin_expr(ident: ast::Ident, is_raw: bool) -> bool {
         keywords::Continue.name(),
         keywords::False.name(),
         keywords::For.name(),
-        keywords::Let.name(),
+        // FIXME(53667): Consider whether `Let` can be added here.
```

Comment:
this will suddenly (without a feature gate) lock us in to changing what counts as an expression in macros?

Reply:
I believe it would, but I had to comment it out in any case due to https://github.com/rust-lang/rust/pull/60861#issuecomment-492813026 (😭!) so this might not be actionable at all; will have to consider what to do but I want to postpone that consideration to after this PR since it's a peripheral problem.

Reply:
`Let` needs to be added to here if it's an expression.

The backward compatibility issue can be addressed for `macro_rules` macros specifically, in `libsyntax/ext/tt/macro_parser.rs`.
`sym::expr => token.can_begin_expr() && token != kw::Let` or something.

Reply:
Ah, thanks! Will try that :)

Reply:
Fixed (locally).